oonib/onion.py

```diff
@@ -1,91 +0,0 @@
-import tempfile
-from oonib import log
-from oonib.config import config
-from twisted.internet import reactor, endpoints
-import os
-
-from random import randint
-import socket
-
-from txtorcon import TCPHiddenServiceEndpoint, TorConfig
-from txtorcon import launch_tor
-
-from txtorcon import __version__ as txtorcon_version
-if tuple(map(int, txtorcon_version.split('.'))) < (0, 9, 0):
-    """
-    Fix for bug in txtorcon versions < 0.9.0 where TCPHiddenServiceEndpoint
-    listens on all interfaces by default.
-    """
-    def create_listener(self, proto):
-        self._update_onion(self.hiddenservice.dir)
-        self.tcp_endpoint = endpoints.TCP4ServerEndpoint(self.reactor,
-                                                         self.listen_port,
-                                                         interface='127.0.0.1')
-        d = self.tcp_endpoint.listen(self.protocolfactory)
-        d.addCallback(self._add_attributes).addErrback(self._retry_local_port)
-        return d
-    TCPHiddenServiceEndpoint._create_listener =  create_listener
-
-def randomFreePort(addr="127.0.0.1"):
-    """
-    Args:
-
-        addr (str): the IP address to attempt to bind to.
-
-    Returns an int representing the free port number at the moment of calling
-
-    Note: there is no guarantee that some other application will attempt to
-    bind to this port once this function has been called.
-    """
-    free = False
-    while not free:
-        port = randint(1024, 65535)
-        s = socket.socket()
-        try:
-            s.bind((addr, port))
-            free = True
-        except:
-            pass
-        s.close()
-    return port
-
-def txSetupFailed(failure):
-    log.err("Setup failed")
-    log.exception(failure)
-
-def configTor(torconfig):
-    def updates(prog, tag, summary):
-        print("%d%%: %s" % (prog, summary))
-
-    if config.main.socks_port:
-        torconfig.SocksPort = config.main.socks_port
-    if config.main.control_port:
-        torconfig.ControlPort = config.main.control_port
-    if config.main.tor2webmode:
-        torconfig.Tor2webMode = 1
```

Comment:
We actually need to set this in order to be able to run our custom compiled version of Tor with the tor2web mode enabled.